### PR TITLE
[REF] Mapping - Check FK on delete with SQL instead of PHP (dev/core#2757)

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Part of an ongoing effort to clean up the DAO classes and use generic functions and events.

Before
----------------------------------------
For some reason, a php function was mimicking ON DELETE CASCADE in mySql.

After
----------------------------------------
Uses actual ON DELETE CASCADE in mySql.

Comments
------------
The only reason I can fathom to do it in PHP rather than in SQL is to call pre and post hooks... which this wasn't.

### CODE REVIEW

File: CRM/Upgrade/Incremental/sql/5.42.alpha1.mysql.tpl [hunk 1]
```diff
@@ -1 +1,5 @@
 {* file to handle db changes in 5.42.alpha1 during upgrade *}
+
+ALTER TABLE `civicrm_mapping_field` DROP FOREIGN KEY `FK_civicrm_mapping_field_mapping_id`;
```
Comment (eileenmcnaughton) [inline CRM/Upgrade/Incremental/sql/5.42.alpha1.mysql.tpl:3]: @seamuslee001 I'm getting confused here - we use `CRM_Core_BAO_SchemaHandler::safeRemoveFK(...)` to drop the index I think but ?? to add one

**Comment by colemanw** [comment]
@eileenmcnaughton @seamuslee001 does this PR require a schema regen?

File: CRM/Upgrade/Incremental/sql/5.42.alpha1.mysql.tpl (hunk 1 above)
Comment (seamuslee001) [reply CRM/Upgrade/Incremental/sql/5.42.alpha1.mysql.tpl:3]: You could use this https://github.com/civicrm/civicrm-core/blob/33f8687f14294e03392b47bec038f747e19ecfda/CRM/Core/BAO/SchemaHandler.php#L256 but still not perfect

**Comment by seamuslee001** [comment]
@colemanw whilst you looking at mapping I'm wondering if we need to modify this https://github.com/civicrm/civicrm-core/blob/master/xml/schema/Contact/SavedSearch.xml#L82 at all

**Comment by colemanw** [comment]
@seamuslee001 why would we need to modify it?

**Comment by seamuslee001** [comment]
@colemanw should it be cascade delete also rather than set null?

**Comment by eileenmcnaughton** [comment]
@colemanw I've pushed up changes to the upgrade script as discussed above

I tested locally upgrading from 5.42 to 5.43 with this version of the upgrade script
I think deleted a Mapping using api explorer (v4) and checked the fields were deleted - all good IMHO

I think @seamuslee001's question about search kit fk on mapping_id is out of scope - from memory it is used in search builder context and I have a feeling there might be some complications in answering the question about whether it should delete on cascade so it's not something we could 'quickly do while we are at it'

This is mergeable now IMHO, Since I altered it I'm added 'merge-ready' not 'merge-on-pass'

**Comment by colemanw** [comment]
Thanks @eileenmcnaughton - this looks good except is this one of those situations where regen.sh has to be run and new sql install files committed?

**Comment by eileenmcnaughton** [comment]
I don't think it does

**Comment by colemanw** [comment]
In that case this is good to go

**Pull request merged**